Project-Descriptions-and-Plans/LLVM-Toolchain/COREV_LLVM_Plan_Approved_2023.md

```diff
@@ -0,0 +1,173 @@
+| PA gate: Plan Approved               | Presented on 2022-11-28 for **2023 workplan**, approved on 2023-xx-xx |
+| PA gate: Plan Approved               | 2024 and beyond workplan in future PA gate |
+
+Author: ChunyuLiao
+
+## Summary of project
+
+This document should be read in the context of the overall CORE-V LLVM project project launch proposal.  It does not duplicate information from that document.
+
+This plan is to deliver Clang/LLVM compiler support for all of the CV32E40P version 2 instruction set extensions.
+
+### Summary of Timeline
+
+CORE-V LLVM project is a long-run project. It is therefore difficult to plan the complete project.
+
+Therefore, in order to keep the project under control, the project plan is defined for 2023 activity. 
+
+## OpenHW Members/Participants committed to participate in CORE-V LLVM project
+
+- Embecosm 
+- Institute of Software Chinese Academy of Sciences (ISCAS/PLCT)
+
+## Project Leader(s)
+
+- Charlie Keaney (overall LLVM project lead)
+- Chunyu Liao (project lead for this project)
+
+## Project Planning Documents
+
+The progress towards the milestones will be tracked during progress meetings, usually every week.
+
+A monthly report will be provided to the Software Task Group.
+
+## Summary of requirements
+
+### CORE-V LLVM project features
+
+After approval by the TWG, it will be managed in OpenHW GitHub.
+
+As a summary, contains:
+
++ Clang/LLVM tool chain for C and C++:
+    + assembler
+    + compiler
+    + support for ISA extensions
+
+The project will use the existing GNU linker and debugger. The standard RISC-V newlib and compilerRT libraries will be used unchanged.
+
+### License scheme
+
+The LLVM project mostly uses Apache licensing. The full details of all licenses is addressed in the Project Launch document.
+
+### Verification
+
+Github Continuous Integration and Test (CI) will be set up.
+
+## Explanation of why OpenHW should do this project
+
+This document should be read in the context of the overall CORE-V LLVM project project launch proposal.  It does not duplicate information from that document.
+
+## Industry landscape: description of competing, alternative, or related efforts in the industry
+This document should be read in the context of the overall CORE-V LLVM project project launch proposal.  It does not duplicate information from that document.
+
+### Market differentiators
+## External dependencies
++ Prequisites:
+    + a suitable platform for regression testing the compiler.
+       * expected to be either OVPSim or the Verilator model of CV32E40Pv2 due in early 2023.
+    + agreement on the instruction set encodings to be compliant with the RISC-V standard.
+       * draft is currently in [this spreadsheet](https://github.com/openhwgroup/cv32e40p/files/9949531/pulp_encoding_blocks-OPHW-2022-10-07.xlsx).
++ External dependencies:
+    + ongoing tracking of upstream LLVM development until the CORE-V tool chain is accepted upstream.
```

Comment:
Is the ongoing tracking mainly someone keeping an eye on the work or is something more specific needed?

Reply:
This is a responsibility of @charkeaney as overall LLVM project lead.